diff --git a/target/linux/airoha/dts/an7581-w1701k.dts b/target/linux/airoha/dts/an7581-w1701k.dts
index fd43178300..4af2f0119e 100644
--- a/target/linux/airoha/dts/an7581-w1701k.dts
+++ b/target/linux/airoha/dts/an7581-w1701k.dts
@@ -200,7 +200,20 @@
 &pcie0 {
 	status = "okay";
 
-	airoha,x2-mode;
+	reg = <0x0 0x1fc00000 0x0 0x1670>,
+	      <0x0 0x1fc20000 0x0 0x1670>;
+	reg-names = "pcie-mac", "sec-pcie-mac";
+
+	resets = <&scuclk EN7581_PCIE0_RST>,
+		 <&scuclk EN7581_PCIE1_RST>,
+		 <&scuclk EN7581_PCIC_PERSTOUT0_RST>,
+		 <&scuclk EN7581_PCIC_PERSTOUT1_RST>;
+	reset-names = "phy-lane0", "phy-lane1",
+		      "perstout", "sec-perstout";
+
+	num-lanes = <2>;
+
+	airoha,scu = <&scuclk>;
 
 	pinctrl-names = "default";
 	pinctrl-0 = <&pcie0_rst_pins>;
@@ -244,6 +257,10 @@
 	};
 };
 
+&pcie1 {
+	status = "disabled";
+};
+
 &npu {
 	status = "okay";
 };
